cumulusci/tasks/debug_logs.py

```diff
@@ -0,0 +1,135 @@
+import datetime
+import os
+import csv
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+
+
+class TurnOnDebugLogs(BaseSalesforceApiTask):
+    task_docs = """
+    Turn on debugging information in Salesforce
+    """
+
+    debug_level_id = None  # Class variable
+
+    def _run_task(self):
+        """Create a TraceFlag for a given user."""
+        self._delete_debug_levels()
+        self._delete_trace_flags()
+        self.logger.info("Creating DebugLevel object")
+        DebugLevel = self._get_tooling_object("DebugLevel")
+        result = DebugLevel.create(
+            {
+                "ApexCode": "Info",
+                "ApexProfiling": "Debug",
+                "Callout": "Info",
+                "Database": "Info",
+                "DeveloperName": "CumulusCI",
+                "MasterLabel": "CumulusCI",
+                "System": "Info",
+                "Validation": "Info",
+                "Visualforce": "Info",
+                "Workflow": "Info",
+            }
+        )
+        if TurnOnDebugLogs.debug_level_id:
+            self.logger.info(
+                "Debug logs are already turned on ({TurnOnDebugLogs.debug_level_id})"
+            )
+            return
+        TurnOnDebugLogs.debug_level_id = result["id"]
+        self.logger.info("Setting up trace flag to capture debug logs")
+        # New TraceFlag expires 12 hours from now
+        expiration_date = datetime.datetime.utcnow() + datetime.timedelta(
+            seconds=60 * 60 * 12
+        )
+        TraceFlag = self._get_tooling_object("TraceFlag")
+        result = TraceFlag.create(
+            {
+                "DebugLevelId": result["id"],
+                "ExpirationDate": expiration_date.isoformat(),
+                "LogType": "USER_DEBUG",
+                "TracedEntityId": self.org_config.user_id,
+            }
+        )
+        self.logger.info("Created TraceFlag for user")
+
+    def _delete_trace_flags(self):
+        """
+        Delete existing DebugLevel objects.
+        This will automatically delete associated TraceFlags as well.
+        """
+        self.logger.info("Deleting existing TraceFlag objects")
+        result = self.tooling.query(
+            "Select Id from TraceFlag Where TracedEntityId = '{}'".format(
```

Comment:
Do we have any formatting best practices when writing SQL/SOQL queries as string in our code? (e.g. all caps for keywords like `SELECT`, `FROM`, `WHERE`)